File: 003- Password Checker/PasswordChecker.cpp
#include <iostream>
#include <cctype>
using namespace std;

bool CheckPassword(const string& str) {
    int n = str.length();
    if (n < 4 || isdigit(str[0])){
        return false;
    } 

    bool hasDigit = false, hasUpper = false;

    for (int i = 0; i < n; i++) {
        char ch = str[i];
        if (isspace(ch) || ch == '/'){
            return false;
        } 
        if (isdigit(ch)){
            hasDigit = true;
        } 
        if (isupper(ch)){
            hasUpper = true;
        }
    }
    return hasDigit && hasUpper;
}

int main() {
    string input;
    cout<<"Enter the password: ";
    getline(cin, input);
    cout << CheckPassword(input) << endl;
    return 0;
}

File: 003- Password Checker/Password Checker.py
def check_password(s: str) -> int:
    if len(s) < 4 or s[0].isdigit():
        return 0

    has_digit = has_upper = False

    for i in range(len(s)):  
        ch = s[i]
        if ch.isspace() or ch == '/':
            return 0
        if ch.isdigit():
            has_digit = True
        if ch.isupper():
            has_upper = True

    return 1 if has_digit and has_upper else 0


print(check_password(input().strip()))
